lib/writeUris.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+var fsExtra = require('fs-extra');
+var mime = require('mime');
+var path = require('path');
+var Promise = require('bluebird');
+
+var fsExtraOutputFile = Promise.promisify(fsExtra.outputFile);
+
+module.exports = writeUris;
+
+/**
+ * Write glTF resources as embedded data uris or external files.
+ *
+ * @param {Object} gltf The glTF asset.
+ * @param {String} gltfPath Path where the glTF will be saved.
+ * @param {Boolean} separateBuffers Writes out separate buffers.
+ * @param {Boolean} separateTextures Writes out separate textures.
+ * @param {Logger} logger A callback function for handling logged messages. Defaults to console.log.
+ * @returns {Promise} A promise that resolves to the glTF asset.
+ *
+ * @private
+ */
+function writeUris(gltf, gltfPath, separateBuffers, separateTextures, logger) {
```

Comment:
This exists so that resources can be embedded or saved externally when using the `--bypassPipeline` flag. It was also added to get around the node string limitation, which is now moot since it throws an error instead (until a better solution is worked out).

Overall though it is still needed here.